test/token/ERC1155/ERC1155Base.behavior.js

```diff
@@ -108,11 +116,39 @@ const describeBehaviorOfERC1155Base = function ({ deploy }, skips) {
     });
 
     describe('#safeTransferFrom', function () {
-      it('todo');
+      it('sends amount from A to B', async function(){
+        const id = ethers.constants.Zero;
+        const amount = ethers.constants.Two;
+
+        expect(await instance.callStatic.balanceOf(holder.address, id)).to.equal(0);
+
+        await mint(holder.address, id, amount);
+
+        expect(await instance.callStatic.balanceOf(holder.address, id)).to.equal(amount);
+
+        await instance.connect(holder).safeTransferFrom(holder.address, spender.address, id, amount, ethers.utils.randomBytes(0));
```

Comment:
Use the `changeTokenBalances` matcher here: https://ethereum-waffle.readthedocs.io/en/latest/matchers.html

No need to check that initial balance is 0.

Reply:
neat, didn't know about that one